binderhub/repoproviders.py

```diff
@@ -692,9 +677,12 @@ def get_repo_url(self):
     @gen.coroutine
     def github_api_request(self, api_url, etag=None):
         client = AsyncHTTPClient()
-        if self.auth:
-            # Add auth params. After logging!
-            api_url = url_concat(api_url, self.auth)
+
+        request_kwargs = {}
+        if self.client_id and self.client_secret:
+            request_kwargs.update(
+                dict(auth_username=self.client_id, auth_password=self.client_secret)
```

Comment:
These keyword arguments are understood by tornado's HTTP client and cause it to do basic auth. leaving it here for the future because I had to double check :)